sys-devel/flang/flang-13.0.0.ebuild

```diff
@@ -0,0 +1,111 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+CMAKE_ECLASS=cmake
+
+PYTHON_COMPAT=( python3_{8..10} )
+inherit cmake-multilib llvm python-single-r1
+
+DESCRIPTION="Flang is a ground-up implementation of a Fortran front end written in modern C++"
+HOMEPAGE=" https://github.com/llvm/llvm-project/tree/main/flang
+		http://flang.llvm.org/docs/"
+SRC_URI="https://github.com/llvm/llvm-project/releases/download/llvmorg-${PV}/llvm-project-${PV}.src.tar.xz
+	https://src.fedoraproject.org/rpms/flang/raw/rawhide/f/0001-Link-against-libclang-cpp.so.patch -> flang-Link-against-libclang-cpp.so.patch
+	https://src.fedoraproject.org/rpms/flang/raw/rawhide/f/0001-PATCH-flang-Disable-use-of-sphinx_markdown_tables.patch -> flang-PATCH-flang-Disable-use-of-sphinx_markdown_tables.patch
+"
+
+LICENSE="Apache-2.0-with-LLVM-exceptions UoI-NCSA MIT"
+SLOT="13"
+IUSE="doc test +clang"
+KEYWORDS="~amd64"
+
+DEPEND="
+	>=sys-devel/clang-13.0.0:13=[static-analyzer,${MULTILIB_USEDEP}]
+	sys-devel/llvm:13=[${MULTILIB_USEDEP}]
+"
+RDEPEND="${DEPEND}
+	${PYTHON_DEPS}"
+
+BDEPEND="
+	doc? (
+		dev-python/sphinx
+		app-doc/doxygen[dot]
+	)
+	test? (
+		dev-python/lit
+	)
+	sys-devel/mlir:13=
+"
+
+REQUIRED_USE="${PYTHON_REQUIRED_USE}"
+RESTRICT="!test? ( test )"
+
+LLVM_MAX_SLOT=13
+
+pkg_setup() {
+	use system-llvm && llvm_pkg_setup
+	if use clang ; then
+		export CC=clang
+		export CXX=clang++
+		export AR=llvm-ar
+		export LD=lld
+		export RANLIB=llvm-ranlib
+		export NM=llvm-nm
+	fi
```

Comment:
Like I said on the other PR, we let users do stuff like this via package.env, not USE flags.

we only do e.g USE=clang if hacks are needed to make things build and we avoid it if possible.

Reply:
I'm afraid that I couldn't quite understand what the problem is. Sorry. 
It is about CC/CXX compiler - it is not recommended to change system wide compiler
or it is about the clang flag - it should be used in different way.

Reply:
If users want to use Clang, they should be using `package.env` e.g. https://wiki.gentoo.org/wiki/Clang#Clang_environments.

We only provide `USE=clang` if certain hacks are needed to make it work (like Firefox).

So you should be able to just remove all of this.

Reply:
Thank you very much. Now I get it what was about. 
Now I understand that my question was quite stupid :)

Reply:
No such thing as a stupid question and you know I'm happy to help you learn! 👍